src/vivarium_gates_mncnh/components/delivery_facility.py

```diff
@@ -50,15 +54,37 @@ def on_time_step(self, event: Event) -> None:
 
         pop = self.population_view.get(event.index)
         # Choose delivery facility type
-        birth_idx = pop.index[
+        not_partial_term = (
             pop[COLUMNS.PREGNANCY_OUTCOME] != PREGNANCY_OUTCOMES.PARTIAL_TERM_OUTCOME
-        ]
-        delivery_facility_type = self.randomness.choice(
-            birth_idx,
-            list(self.delivery_facility_probabilities.keys()),
-            p=list(self.delivery_facility_probabilities.values()),
-            additional_key="delivery_facility_type",
         )
-        pop.loc[birth_idx, COLUMNS.DELIVERY_FACILITY_TYPE] = delivery_facility_type
+        pop = pop[not_partial_term]
+        propensity = self.propensity(pop.index)
+        is_believed_preterm = pop[COLUMNS.STATED_GESTATIONAL_AGE] < 37
+
+        # home births for lower propensities, facility for higher
+        assigned_home_preterm = (
+            propensity[is_believed_preterm]
+            < self.delivery_facility_probabilities[FACILITY_CHOICE.P_HOME_PRETERM]
+        )
```

Comment:
Can you explain what this is doing? I would expect to see something like
`if is_believed_preterm:
    assigned_home_preterm = (propensity[delivery_facility] < self.delivery_facility_probabilities[FACILITY_CHOICE.P_HOME_PRETERM]
else:
    assigned_home_preterm = (propensity[delivery_facility] < self.delivery_facility_probabilities[FACILITY_CHOICE.P_HOME_FULL_TERM]
`
Does that make sense?